src/commands/publish.ts

```diff
@@ -33,56 +40,217 @@ const publish = async (
+        const isBranch = await branchLookup(`v${version}`);
+
+        if (isBranch) {
+            throw new Error(
+                `v${version} already exists, please checkout v${version} branch to publish a new major version`
+            );
+        }
+
+        const res: AxiosResponse = await createMajorBlockRequest(
+            blockData,
+            id,
+            version
+        );
+
+        await createBranch(`v${version}`);
+
+        updateBlockSettingsFile({
+            activeVersion: version,
+            isReleased: false,
+        });
+
+        await updateBranch(`v${version}`);
+
+        logSuccess(`
+            Published ${displayName} v${version} for staging
+            ID ${res.data.id}
+        `);
+
+        exit(0);
+    } catch (err) {
+        logError(err);
+        checkErrorCode(err);
         exit(1);
-        return;
     }
+};
 
-    const publicFlag = togglePublic ? !isPublic : isPublic;
+const update = async (togglePublic: boolean): Promise<void> => {
+    validateFilesExistOrExit();
+    validateBlockExistOrExit();
 
-    updateBlockRequest(
-        { displayName, publishedName },
-        blockData,
+    const filePath = resolve(cwd(), BUILT_FILE_PATH);
+    const blockData = readFileSync(filePath).toString();
+    const {
+        activeVersion,
+        displayName,
         id,
-        publicFlag
-    )
-        .then((res: AxiosResponse) => {
-            updateBlockSettingsFile({ id: res.data.id, isPublic: publicFlag });
-            logSuccess(`Updated ${displayName} with ID ${res.data.id}.`);
-            exit(0);
-        })
-        .catch((err: AxiosError) => {
-            logError(err);
-            checkErrorCode(err);
-            exit(1);
+        isPublic,
+        publishedName,
+    } = readBlockSettingsFile(BLOCK_SETTINGS_FILE);
+
+    const publicFlag = togglePublic ? !isPublic : isPublic;
+
+    try {
+        const res: AxiosResponse = await updateBlockRequest(
+            { displayName, publishedName },
+            blockData,
+            id,
+            publicFlag,
+            activeVersion
+        );
+
+        updateBlockSettingsFile({
+            isPublic: publicFlag,
+            isReleased: false,
         });
+
+        await updateBranch(`v${activeVersion}`);
+
+        logSuccess(`
+            Updated ${displayName} v${activeVersion} for staging
+            ID ${res.data.id}
+        `);
+
+        exit(0);
+    } catch (err) {
+        logError(err);
+        checkErrorCode(err);
+        exit(1);
+    }
+};
+
+const release = async (note: string): Promise<void> => {
+    validateFilesExistOrExit();
+    validateBlockExistOrExit();
+
+    const { activeVersion, displayName, id } = readBlockSettingsFile(
+        BLOCK_SETTINGS_FILE
+    );
+
+    try {
+        const res: AxiosResponse = await releaseBlockRequest(
+            id,
+            note,
+            activeVersion
+        );
+
+        updateBlockSettingsFile({
+            isReleased: true,
```

Comment:
I'm a bit worried on relying on this flag  (and others) in the block file for our logic. Are we safe if users manually modify this file?

Reply:
I actually don't need that flag anymore, going to remove it.